src/balloontoolbar.js

```diff
@@ -0,0 +1,92 @@
+/**
+ * @license Copyright (c) 2003-2017, CKSource - Frederico Knabben. All rights reserved.
+ * For licensing, see LICENSE.md.
+ */
+
+/**
+ * @module editor-balloon-toolbar/contextual
+ */
+
+import StandardEditor from '@ckeditor/ckeditor5-core/src/editor/standardeditor';
+import HtmlDataProcessor from '@ckeditor/ckeditor5-engine/src/dataprocessor/htmldataprocessor';
+import ContextualToolbar from '@ckeditor/ckeditor5-ui/src/toolbar/contextual/contextualtoolbar';
+import BalloonToolbarEditorUI from './balloontoolbareditorui';
+import BalloonToolbarEditorUIView from './balloontoolbareditoruiview';
+
+import '../theme/theme.scss';
+
+/**
+ * The balloon toolbar editor. Uses an inline editable and a toolbar based
+ * on the {@link ui/toolbar/contextual/contextualtoolbar~ContextualToolbar}.
+ *
+ * @extends module:core/editor/standardeditor~StandardEditor
+ */
+export default class BalloonToolbarEditor extends StandardEditor {
+	/**
+	 * Creates an instance of the balloon toolbar editor.
+	 *
+	 * @param {HTMLElement} element The DOM element that will be the source for the created editor.
+	 * @param {Object} config The editor configuration.
+	 */
+	constructor( element, config ) {
+		super( element, config );
+
+		this.config.get( 'plugins' ).push( ContextualToolbar );
+		this.config.define( 'contextualToolbar', this.config.get( 'toolbar' ) );
```

Comment:
Well... this is quite a story. At the very beginning, this editor type expected developers to configure it like this:
```js
BalloonToolbarEditor.create( editorElement, {
	plugins: [ ContextualToolbar ],
	contextualToolbar: [ ... ]
} );
```
but in a F2F talk with @Reinmar we found out this is not a very straightforward since editor-inline and editor-classic don't require a) any special plugins b) custom config names (`toolbar` vs `contextualToolbar`). 

So we decided to hack things a little bit to load `ContextualToolbar` automatically and create a "bridge" between `cfg.toolbar` and `cfg.contextualToolbar` to keep the API between the creators as similar as possible.

> so further changes to toolbar configuration will also affect contextualToolbar

Well, I suppose it doesn't matter. Once you decide to use `BalloonToolbarEditor`, it takes the full control over `ContextualToolbar`. I don't see how `ContextualToolbar` could remain useful for other features once seized by the `BalloonToolbarEditor` 🤔 .